### PR TITLE
✨ Nextcloud 16 polishing

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Add eslint with the same rules as in other apps
---
- [x] Dismissing older notification also hides every newer notification (only until refresh)
- [x] Notifications window should have min-height (same as when it’s empty)
- [x] Needs short compress animation for dismissing for proper feedback

### CODE REVIEW

**Comment by nickvergessen** [comment]
>  Needs short compress animation for dismissing for proper feedback

Can a @nextcloud/javascript  do this or tell me how? I guess it goes here and fade is not enough/working? https://github.com/nextcloud/notifications/blob/c349dccbcf27ba041843e206b843fb18d0d0223a/src/components/Notification.vue#L213

> Dismissing older notification also hides every newer notification (only until refresh)

Although the logic is the same as in all other apps I used Vue in until now, index is undefined in https://github.com/nextcloud/notifications/blob/c349dccbcf27ba041843e206b843fb18d0d0223a/src/App.vue#L118 removing always the first item, so when you delete teh second one of two, the list shows nothing anymore.

**Comment by MorrisJobke** [comment]
> removing always the first item, so when you delete teh second one of two, the list shows nothing anymore.

FYI: Having 3 notifications then deleting the second results in a list with only the third one.

**Comment by nickvergessen** [comment]
@skjnldsv 
* Fixed position of empty content
* Transition in chromium on last removal

**Comment by skjnldsv** [comment]
DOne

File: src/App.vue [hunk 1]
```diff
@@ -1,200 +1,232 @@
...
+						<notification
+							v-for="(n, index) in notifications"
+							:key="n.notification_id"
+							v-bind="n"
+							:index="index"
+							:notification-id="n.notification_id"
+							:object-id="n.object_id"
+							:object-type="n.object_type"
+							@remove="onRemove" />
+						<div v-if="notifications.length > 2" class="dismiss-all" @click="onDismissAll">
```
Comment (nickvergessen) [inline src/App.vue]: needs to be li? @skjnldsv

Comment (nickvergessen) [reply src/App.vue]: Ah no, understood it now.

File: src/App.vue [hunk 2]
```diff
@@ -1,200 +1,232 @@
...
-	import notification from './components/notification';
-	import axios from 'nextcloud-axios';
+import Notification from './components/Notification'
+import axios from 'nextcloud-axios'
+import _ from 'lodash'
 
-	export default {
-		name: 'app',
+export default {
+	name: 'App',
```
Comment (skjnldsv) [inline src/App.vue]:
```suggestion
	name: 'NotificationsList',
```

File: src/init.js [hunk 3]
```diff
@@ -18,17 +18,27 @@
...
+Array.prototype.map.call(searchBox, (el) => {
+	if (el.nodeName !== 'FORM') {
+		return
+	}
+
+	el.insertAdjacentHTML('afterend', notificationsBell.outerHTML)
+})
+
+export default new Vue({
 	el: '#notifications',
```
Comment (skjnldsv) [inline src/init.js]:
```suggestion
	el: '#notifications',
	name: 'NotificationsRoot',
```

File: src/App.vue [hunk 4]
```diff
@@ -1,200 +1,232 @@
...
 	}
+}
 </script>
+
+<style scoped>
+	.fade-enter-active,
+	.fade-leave-active,
+	.fade-collapse-enter-active,
+	.fade-collapse-leave-active {
+		transition: opacity 200ms, max-height 200ms;
```
Comment (jancborchardt) [inline src/App.vue]: Best use `var(--animation-quick)` here, which translates to 100ms. :)

Comment (jancborchardt) [inline src/App.vue]:
```suggestion
		transition: opacity var(--animation-quick), max-height var(--animation-quick);
```

Comment (jancborchardt) [reply src/App.vue]: Why is this resolved @nickvergessen, shouldn't the change be applied or does it not work? :)

Comment (nickvergessen) [reply src/App.vue]: Seems i forgot to pull after merging this in the ui

**Pull request merged**